analytics/analytics/models.py

```diff
@@ -66,3 +66,23 @@ class Meta:
         constraints = [
             models.UniqueConstraint(fields=["path", "timer"], name="unique-phase-path"),
         ]
+
+
+class ErrorTaxonomy(models.Model):
+    created = models.DateTimeField(auto_now_add=True)
+    job = models.OneToOneField(
+        Job, related_name="error_taxonomy", on_delete=models.CASCADE
+    )
+
+    attempt_number = models.PositiveSmallIntegerField()
+    retried = models.BooleanField()
+    kubernetes_job = models.BooleanField(
+        help_text="True if this job is a Kubernetes job, False otherwise."
+    )
+
+    error_taxonomy = models.CharField(max_length=64)
+    error_taxonomy_version = models.CharField(max_length=32)
+
+    webhook_payload = models.JSONField(
+        help_text="The JSON payload received from the GitLab job webhook."
+    )
```

Comment:
These webhooks payloads are decently large, correct? I'm somewhat concerned about storing lots of data here, but I don't think we should block on this. Just might be something to keep an eye on.

Reply:
They're not small, but I don't think they're anything too big for postgres (especially relative to the amount of timing data we're already storing). Here's an example of one https://docs.gitlab.com/ee/user/project/integrations/webhook_events.html#job-events

Reply:
Yeah, I think we should just keep an eye on it. We only have ~550k records in the job table atm, so the amount of data this is likely to represent isn't huge anyway, but it's definitely dwarfing most other fields we store.